### PR TITLE
Fixes #283

### LINKED ISSUE #283: `vmime::text::createFromString()` drops spaces if an 8-bit word is followed by a 7-bit word
Following example

```
vmime::mailbox mailbox(vmime::text("Test München West", vmime::charsets::WINDOWS_1252), "test@example.com");
vmime::utility::outputStreamAdapter adapter(std::cout);
mailbox.generate(adapter);
```

gives as output

```
=?us-ascii?Q?Test_?= =?windows-1252?Q?M=FCnchen?= =?us-ascii?Q?West?=
  <test@example.com>
```

The first space between `Test` and `München` is encoded as an underscore along with the first word: `Test_.` The second space between `München` and `West` is encoded with neither of the two words and thus lost. Decoding the text results in `Test MünchenWest` instead of `Test München West`.

This is caused by the way `vmime::text::createFromString()` handles transitions between 7-bit and 8-bit words:
https://github.com/kisli/vmime/blob/1a35bb6d71b6301287e21aaabd112997ea0f0a7f/src/vmime/text.cpp#L310-L346
If an 8-bit word follows a 7-bit word, a space is appended to the previous word (lines 321 - 324). The opposite case of a 7-bit word following an 8-bit word misses this behaviour (lines 339- 345). Pull request https://github.com/kisli/vmime/pull/284 fixes this.

### PR DESCRIPTION
**Pull request by RichardSteele**
Prevents the loss of a space if the previous word is 8-bit but the current one is not. This equals the case of the current word being 8-bit but not the previous one.

### CODE REVIEW

**Comment by vincent-richard** [comment]
Hello!

Unfortunately, the patch you proposed introduces a regression and breaks the "parser_textTest" unit test, at 3 places:

1) test: textTest::testNewFromString (F) line: 182
2) test: textTest::testBugFix20110511 (F) line: 593
3) test: textTest::testInternationalizedEmail_whitespace (F) line: 711

**Comment by RichardSteele** [comment]
Right, I forgot about the tests.

I think this bug can't be resolved without breaking other parts. As per [RFC2047](https://datatracker.ietf.org/doc/html/rfc2047) spaces between adjacent encoded words are just separators but not meant to be displayed. A space between an encoded word and a regular ASCII text is not just a separator but also meant to be displayed.

The problem is that `vmime::text::createFromString()` doesn't know whether or not the text will be forcefully encoded later on, which will create encoded words even for ASCII texts, like in a mailbox field. Likewise, during the stringification of a text it's unclear whether or not the text was created manually or by `createFromString()`. Handling the bug at this point could end up in adding redundant spaces.

As [RFC2047](https://datatracker.ietf.org/doc/html/rfc2047) says:

> Use of 'encoded-word's to represent strings of purely ASCII characters is allowed, but discouraged.

In conclusion, it might be better to put a warning in the comments of `vmime::text`.
